tests/js/server/replication2/replication2-supervision-replicated-state-target-cluster.js

```diff
@@ -179,6 +178,49 @@ const replicatedStateSuite = function () {
         [newParticipant]: {allowedInQuorum: true, allowedAsLeader: true, forced: false},
       }));
     },
+
+    testSetLeader: function () {
+      const {stateId, followers} = createReplicatedState();
+      const newLeader = _.sample(followers);
+      updateReplicatedStateTarget(database, stateId,
+        (target) => {
+          target.leader = newLeader;
+          return target;
+        });
+      lh.waitFor(() => {
+        const currentLeader = lh.getReplicatedLogLeaderTarget(database, stateId);
+        if (currentLeader === newLeader) {
+          return true;
+        } else {
+          return new Error(`Expected log leader to switch to ${newLeader}, but is still ${currentLeader}`);
+        }
+      });
+      lh.waitFor(() => {
+        const {leader: currentLeader} = lh.getReplicatedLogLeaderPlan(database, stateId);
+        if (currentLeader === newLeader) {
+          return true;
+        } else {
+          return new Error(`Expected log leader to switch to ${newLeader}, but is still ${currentLeader}`);
+        }
+      });
```

Comment:
Would it make sense to put this into the helper library as commonly used predicates?